### PR TITLE
[adc_ctrl] updated the markdown file 

### PR DESCRIPTION
**Pull request by esshiu**
Signed-off-by: Eric Shiu <eshiu@google.com>

### CODE REVIEW

**Review by msfschaffner — COMMENTED** [review]
Thanks @esshiu. Can you try to debug the documentation build locally? This seems to fail CI...
`util/build_docs.py --preview`

As for the D2 change, please put this into a separate commit and fill out the checklist. 
See here how this is done, for example: https://github.com/lowRISC/opentitan/pull/5892/commits/5571f50fe23f1eab0433a7c13e8dae75a49fc295

**Comment by esshiu** [comment]
Cleaned up the docs. Will move the D2 to another commit.

-Eric

File: hw/ip/adc_ctrl/doc/_index.md [hunk 1]
```diff
@@ -1,34 +1,215 @@
...
+title: "Analog to Digital Converter Control Interface"
 ---
 
 # Overview
 
-
+This document specifies the ADC controller IP functionality.
+This IP block implements control and filter logic for an analog block that implements a dual ADC.
+This module conforms to the [Comportable guideline for peripheral functionality.]({{< relref "doc/rm/comportability_specification" >}})
+See that document for integration overview within the broader top level system
```
Comment (msfschaffner) [inline hw/ip/adc_ctrl/doc/_index.md:10]: Nit: missing period at the end of the sentence.

File: hw/ip/adc_ctrl/doc/_index.md [hunk 2]
```diff
@@ -1,34 +1,215 @@
...
 ## Features
 
+The IP block implements the following features:
 
+- Register interface to dual ADC analog block
+- Support for 2 ADC channels
+- Support for 8 filters on the values from the channels
+- Support ADCs with 10-bit output (two reserved bits in CSR)
+- Support for debounce timers on the filter output
+- Uses slow clock to enable use while the part is sleeping
```
Comment (msfschaffner) [inline hw/ip/adc_ctrl/doc/_index.md:21]: Maybe rephrase as `Runs on a slow always-on clock to enable usage while the device is sleeping`

File: hw/ip/adc_ctrl/doc/_index.md [hunk 3]
```diff
@@ -1,34 +1,215 @@
...
+- Low power periodic scan mode for monitoring ADC channels
 
 ## Description
 
+The ADC controller is a simple front-end to an analog block that allows filtering and debouncing of the analog signals
```
Comment (msfschaffner) [inline hw/ip/adc_ctrl/doc/_index.md:26]: Nit: missing period at the end of the sentence. 

(there are several other sentences below that miss a period at the end)

File: hw/ip/adc_ctrl/doc/_index.md [hunk 4]
```diff
@@ -1,34 +1,215 @@
...
+## Sampling state machine
 
-# Programmers Guide
+The state machine that takes ADC samples follows a very simple pattern:
+
+1. *Power up ADC*: The controller issues the power up command to the ADC.
+
+2. *Wait for ADC turn on*: The controller waits for the number of clock cycles programmed in {{< regref "adc_pd_ctl.pwrup_time" >}} which should be set to match the ADC power up delay.
+
+3. *Take sample Channel 0*: The ADC is requested to sample channel 0. When the ADC signals complete the value is stored in {{< regref "adc_chn_val[0].adc_chn_value" >}}. Note that the time taken in this step depends on the properties of the ADC.
```
Comment (msfschaffner) [inline hw/ip/adc_ctrl/doc/_index.md:54]: Nit: put each sentence on a new line, e.g. as you have done for point 6. below (this should still group the text below a bullet point).

File: hw/ip/adc_ctrl/doc/_index.md [hunk 5]
```diff
@@ -1,34 +1,215 @@
...
+
+5. *Evaluate Filters*: The filters are evaluated and debounce logic applied (see [next section](#filters-and-debounce).
+
+6. *Scan type check*: At this point if the {{< regref "adc_pd_ctl.lp_mode" >}} bit is clear scanning continues at step (3).
+   If the bit is set the next step depends on how many samples have hit the filters.
+   If more than {{< regref "adc_lp_sample_ctl.lp_sample_cnt" >}} samples have hit then continuous scanning continues at step (3) else periodic scanning will continue at the next step (7).
+
+7. *Power off ADC*: The controller issues the power down command to the ADC.
+
+8. *Wait sleep time*: the controller will wait for the next sample timer to timeout before restarting at step (1).
```
Comment (msfschaffner) [inline hw/ip/adc_ctrl/doc/_index.md:66]: Nit: Capitalize *T*he for consistency with other bullet points

File: hw/ip/adc_ctrl/doc/_index.md [hunk 6]
```diff
@@ -1,34 +1,215 @@
...
+
+In active operation the controller is in continuous scanning mode:
+* The ADC is continually powered on.
+* The sampling cycle time is the time taken for the ADC to take two samples(450us) plus NNN (fill in based on implementation but probably 4-10)) number of slow clock cycles x (5us; always-on clock frequency is 200KHz).
```
Comment (msfschaffner) [inline hw/ip/adc_ctrl/doc/_index.md:70]: nit: missing space between `samples(450us)`

is `NNN` known by now?

File: hw/ip/adc_ctrl/doc/_index.md [hunk 7]
```diff
@@ -1,34 +1,215 @@
...
+
+Although it can be used at any time, the periodic operation mode and use of the slow clock allows the ADC controller to continue to scan when most of the chip is in sleep or power-down modes.
+The controller can be configured to issue a wakeup to the rest of the chip.
+In this case if any filter enabled in the {{< regref "adc_wakeup_ctl" >}} register is hit then the wakeup will be issued.
+The filter that hit is recorded in the  {{< regref "adc_wakeup_status" >}} register and the bit should be cleared to acknowledge the wake has happened (no further wake events will be issued until this is done).
+
+
+## Filters and debounce
+
+There are two reserved bits in ADC filter control registers for future use. In the current implementation, ADC has 10-bit granularity. Each step is 2.148mV. It covers 0-2.2V.
```
Comment (msfschaffner) [inline hw/ip/adc_ctrl/doc/_index.md:90]: Nit: One sentence per line

File: hw/ip/adc_ctrl/doc/_index.md [hunk 8]
```diff
@@ -1,34 +1,215 @@
...
+* Inside `min=7`, `max=23`: any value between and including 7 and 23 will hit.
+* Outside `min=7`, `max=23`: any value less than 7 or greater than 23 will hit.
+* Inside `min=7`, `max=7`: the value must be exactly 7 to hit (sample noise may make an exact hit unlikely).
+* Inside `min=0`, `max=7`: the value must be less than 8 to hit.
+* Outside `min=8`, `max=0xFFF`: the value must be less than 8 to hit (alternate method).
+* Inside `min=0`, `max=0xFFF`: any value will hit. This may be useful to exclude one channel from the filter.
+* Outside `min=0`, `max=0xFFF`: no value will hit. If set for either channel the filter is effectively disabled.
+
+All pairs of filters that are enabled in {{< regref "adc_chn0_filter_ctl[7:0]" >}} and {{< regref "adc_chn1_filter_ctl[7:0]" >}} are evaluated after each pair of samples has been taken.
+The result is passed to the appropriate counter: the periodic scan counter if enabled and not at its limit otherwise the debounce counter.
```
Comment (msfschaffner) [inline hw/ip/adc_ctrl/doc/_index.md:109]: Nit:
I would insert some commas here or maybe slightly rephrase this sentence as:
```
The filter result is passed to the periodic scan counter, if it is enabled and not at its limit. 
Otherwise the result is passed to the debounce counter.

The list below describes how the counters interpret the filter results:
```

**Comment by msfschaffner** [comment]
Thanks Eric, just a few small comments.
Also, I have adjusted the PR title accordingly (removed D2).

File: hw/ip/adc_ctrl/doc/_index.md (hunk 1 above)
Comment (esshiu) [reply hw/ip/adc_ctrl/doc/_index.md:10]: Sure.

File: hw/ip/adc_ctrl/doc/_index.md (hunk 2 above)
Comment (esshiu) [reply hw/ip/adc_ctrl/doc/_index.md:21]: Sure.

File: hw/ip/adc_ctrl/doc/_index.md (hunk 4 above)
Comment (esshiu) [reply hw/ip/adc_ctrl/doc/_index.md:54]: Sure.

File: hw/ip/adc_ctrl/doc/_index.md (hunk 5 above)
Comment (esshiu) [reply hw/ip/adc_ctrl/doc/_index.md:66]: Sure

File: hw/ip/adc_ctrl/doc/_index.md (hunk 6 above)
Comment (esshiu) [reply hw/ip/adc_ctrl/doc/_index.md:70]: I don't know the NNN cycles yet. Waiting for the DV environment,

File: hw/ip/adc_ctrl/doc/_index.md (hunk 7 above)
Comment (esshiu) [reply hw/ip/adc_ctrl/doc/_index.md:90]: Sure.

File: hw/ip/adc_ctrl/doc/_index.md (hunk 8 above)
Comment (esshiu) [reply hw/ip/adc_ctrl/doc/_index.md:109]: Sure.

**Comment by esshiu** [comment]
Thank you, Michael for the suggested texts.

**Comment by msfschaffner** [comment]
LGTM

@tjaychen WDYT, can we merge this?

**Review by msfschaffner — APPROVED** [review]

**Comment by tjaychen** [comment]
sounds good let's merge.
